### ISSUE TITLE
[clang-tidy] False positive `cppcoreguidelines-rvalue-reference-param-not-moved` when constructor is deleted

### ISSUE THREAD

**Issue by ChrisThrasher** [llvm/llvm-project#69412]
https://godbolt.org/z/dnE37P4c4

```cpp
struct S
{
    S(const int&);
    S(int&&) = delete;
};
```

SFML uses this pattern where `const&` overloads are kept while `&&` overloads are deleted to prevent constructor from temporary values. The `&&` overloads get flagged by `cppcoreguidelines-rvalue-reference-param-not-moved` in spite of them being deleted and thus having no function body.
